packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts

```diff
@@ -211,6 +211,11 @@ export interface IDatePickerProps extends IBaseProps<IDatePicker>, React.HTMLAtt
    * Whether the CalendarDay close button should be shown or not.
    */
   showCloseButton?: boolean;
+
+  /**
+   * DOM id to tag the input label with, for reference.
+   */
+  labelId?: string;
```

Comment:
I don't think `labelId` is the right name... `id` has uses for more than for `label` and this overspecializes the `id`. 

Separately, I suspect there's probably a pattern for doing scenarios like this. @dzearing might be able to advise or we can try looking at a few other components to see how they solve something like this...

Reply:
Hm I think the way Dropdown does this makes a lot more sense and is a more elegant manner to solve what you're trying to do here. Could you do something similar?

Reply:
@dzearing Cliff and I chatted about this a bit. I could also use the native `id` prop and append `-label` internally, but then that requires consumers of `DatePicker` to know that they have to append `-label` to their prop-provided `id` value when using `htmlFor`. That just seems a bit odd to me since it's not part of an API contract and a consumer wouldn't necessarily know about this suffix or what it is without looking at code.